src/lib/tray.ts

```diff
@@ -28,48 +31,72 @@ export const startFlowTimer = async (startTime: DateTime) => {
   }
 
   if (!tray) return
-  await tray.setIcon(null)
-  
-  let currentDuration = useFlowTimer.getState().duration
-  
-  const unsubscribe = useFlowTimer.subscribe(
-    (state) => {
-      currentDuration = state.duration
-    }
-  )
+  await tray.setTitle('')
   
   timerInterval = setInterval(async () => {
-    let timerDuration: Duration
-    if(!currentDuration) {
-      timerDuration = startTime.diffNow().negate()
+    const currentTotalDuration = useFlowTimer.getState().totalDuration
+
+    let formattedTime: string
+    let elapsedMs: number
+    let totalMs: number
+    let shouldStop = false
+
+    if (currentTotalDuration) {
+      const elapsedDuration = startTime.diffNow().negate()
+      let remainingDuration = currentTotalDuration.minus(elapsedDuration)
+
+      if (remainingDuration.as('milliseconds') < 0) {
+        remainingDuration = Duration.fromMillis(0)
+      }
+
+      formattedTime = remainingDuration.toFormat('hh:mm:ss')
+      elapsedMs = elapsedDuration.as('milliseconds')
+      totalMs = currentTotalDuration.as('milliseconds')
+
+      if (remainingDuration.as('milliseconds') <= 0) {
+        shouldStop = true
+      }
     } else {
-      const timeSinceStart = startTime.diffNow()
-      timerDuration = currentDuration.plus(timeSinceStart)
+      const elapsedDuration = startTime.diffNow().negate()
+      formattedTime = elapsedDuration.toFormat('hh:mm:ss')
+      elapsedMs = elapsedDuration.as('milliseconds')
+      totalMs = 0
+    }
+
+    try {
+      const iconBytes = await invoke<Uint8Array>('generate_timer_icon', {
+        timeString: formattedTime,
+        currentMs: elapsedMs,
+        totalMs: totalMs
+      })
+      const iconImage = await Image.fromBytes(iconBytes)
+      await tray.setIcon(iconImage)
+    } catch (error) {
+      logError(`Error generating/setting timer icon: ${error}`)
+    }
+
+    if (shouldStop) {
```

Comment:
Instead of storing a value for shouldStop, I'd just make a function called like "stopTimer" and call that function from the code block where shouldStop is set to true